mycelium: 0.6.1 -> 0.7.1

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.

meta.description for mycelium is: End-2-end encrypted IPv6 overlay network

meta.homepage for mycelium is: https://github.com/threefoldtech/mycelium

meta.changelog for mycelium is: https://github.com/threefoldtech/mycelium/blob/v0.6.2/CHANGELOG.md


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes

- [Release on GitHub](https://github.com/threefoldtech/mycelium/releases/tag/v0.6.2)

- [Compare changes on GitHub](https://github.com/threefoldtech/mycelium/compare/v0.6.1...v0.6.2)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0.6.2 with grep in /nix/store/sfxdfwg6llga5d57qws54snvbjxybvkc-mycelium-0.6.2
- found 0.6.2 in filename of file in /nix/store/sfxdfwg6llga5d57qws54snvbjxybvkc-mycelium-0.6.2

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

mycelium
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/sfxdfwg6llga5d57qws54snvbjxybvkc-mycelium-0.6.2 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A mycelium https://github.com/r-ryantm/nixpkgs/archive/240619e1765b5598b50dc2ee4c27e8d8691c6928.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/240619e1765b5598b50dc2ee4c27e8d8691c6928#mycelium
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/sfxdfwg6llga5d57qws54snvbjxybvkc-mycelium-0.6.2
ls -la /nix/store/sfxdfwg6llga5d57qws54snvbjxybvkc-mycelium-0.6.2/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`
Commit: `240619e1765b5598b50dc2ee4c27e8d8691c6928`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mycelium</li>
  </ul>
</details>

---

###### Maintainer pings

cc @flokli @MatthewCroughan @R-VdP for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Mic92 commented:
We should probably check the NixOS module:

```
This release changes the default location of the private key used to derive the local IP address. If you upgrade to this version and want to keep your IP/subnet, and don't set the -k/--key-file flag, move your key file to the new default location or add the flag pointing to your existing key file.
```

r-ryantm committed:
mycelium: 0.6.1 -> 0.7.1

flokli committed:
nixos/mycelium: update keyFile description

Our location isn't the default location anymore, but makes more sense in
the NixOS module case.
